app/src/main/java/com/example/taskyy/ui/screens/ReminderScreen.kt

```diff
@@ -305,40 +304,46 @@ fun DateSection(
 @Composable
 fun TimeAndDateRow(
     dateString: String,
-    onTimeSelected: (Any?) -> Unit,
-    onDateSelected: (Any?) -> Unit,
+    onDateExpanded: (Boolean) -> Unit,
+    onTimeExpanded: (Boolean) -> Unit,
     isDatePickerExpanded: Boolean,
     isTimePickerExpanded: Boolean,
-    selectedTime: String,
+    timeString: String,
 ) {
     Row(
         verticalAlignment = Alignment.CenterVertically,
         modifier = Modifier.padding(start = 20.dp)
     ) {
         TimeSection(
             isTimePickerExpanded = isTimePickerExpanded,
-            onEvent = onTimeSelected,
-            selectedTime = selectedTime
+            selectedTime = timeString,
+            toggleTimePicker = {
+                onTimeExpanded(it)
+            },
+            userSelectedTime = {
+                ReminderEvent.TimeSelected(selectedTime = it)
```

Comment:
Here you're not responding to the lambda, you just create the event instance which doesn't do anything at this point. That's why your state isn't updating